File: src/main/kotlin/sunsetsatellite/lang/lox/FunctionModifier.kt
package sunsetsatellite.lang.lox

enum class FunctionModifier {
    NORMAL,
    INIT,
    STATIC,
    ABSTRACT,
    NATIVE;


    companion object {
        fun get(token: Token?): FunctionModifier {
            return when (token?.type) {
                TokenType.STATIC -> STATIC
                TokenType.NATIVE -> NATIVE
                else -> NORMAL
            }
        }
    }
}

File: src/main/csharp/SunliteSharp/Core/Modifier/FunctionModifier.cs
using SunliteSharp.Core.Enum;

namespace SunliteSharp.Core.Modifier;

public enum FunctionModifier
{
    Normal,
    Init,
    Static,
    Abstract,
    Native,
    StaticNative,
}

public static class FunctionModifierExtensions
{
    public static FunctionModifier Get(Token? token, Token? token2 = null)
    {
        return token?.Type switch
        {
            TokenType.Static when token2?.Type == TokenType.Native => FunctionModifier.StaticNative,
            TokenType.Static => FunctionModifier.Static,
            TokenType.Native => FunctionModifier.Native,
            _ => FunctionModifier.Normal
        };
    }
}